File: SDKs/amazonaws.com/kinesisanalyticsv2/2018-05-23/go/pkg/models/operations/deleteapplicationinputprocessingconfiguration.go
package operations

import (
"openapi/pkg/models/shared")


type DeleteApplicationInputProcessingConfigurationXAmzTargetEnum string

const (
    DeleteApplicationInputProcessingConfigurationXAmzTargetEnumKinesisAnalytics20180523DeleteApplicationInputProcessingConfiguration DeleteApplicationInputProcessingConfigurationXAmzTargetEnum = "KinesisAnalytics_20180523.DeleteApplicationInputProcessingConfiguration"
)


type DeleteApplicationInputProcessingConfigurationHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget DeleteApplicationInputProcessingConfigurationXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type DeleteApplicationInputProcessingConfigurationRequest struct {
    Headers DeleteApplicationInputProcessingConfigurationHeaders 
    Request shared.DeleteApplicationInputProcessingConfigurationRequest `request:"mediaType=application/json"`
    
}

type DeleteApplicationInputProcessingConfigurationResponse struct {
    ConcurrentModificationException *interface{} 
    ContentType string 
    DeleteApplicationInputProcessingConfigurationResponse *shared.DeleteApplicationInputProcessingConfigurationResponse 
    InvalidArgumentException *interface{} 
    InvalidRequestException *interface{} 
    ResourceInUseException *interface{} 
    ResourceNotFoundException *interface{} 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/kinesisanalyticsv2/2018-05-23/python/src/sdk/models/operations/deleteapplicationinputprocessingconfiguration.py
from dataclasses import dataclass, field
from typing import Any,Enum,Optional
from sdk.models import shared

class DeleteApplicationInputProcessingConfigurationXAmzTargetEnum(str, Enum):
    KINESIS_ANALYTICS_20180523_DELETE_APPLICATION_INPUT_PROCESSING_CONFIGURATION = "KinesisAnalytics_20180523.DeleteApplicationInputProcessingConfiguration"


@dataclass
class DeleteApplicationInputProcessingConfigurationHeaders:
    x_amz_algorithm: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Algorithm' }})
    x_amz_content_sha256: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Content-Sha256' }})
    x_amz_credential: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Credential' }})
    x_amz_date: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Date' }})
    x_amz_security_token: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Security-Token' }})
    x_amz_signature: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Signature' }})
    x_amz_signed_headers: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-SignedHeaders' }})
    x_amz_target: DeleteApplicationInputProcessingConfigurationXAmzTargetEnum = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Target' }})
    

@dataclass
class DeleteApplicationInputProcessingConfigurationRequest:
    headers: DeleteApplicationInputProcessingConfigurationHeaders = field(default=None)
    request: shared.DeleteApplicationInputProcessingConfigurationRequest = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class DeleteApplicationInputProcessingConfigurationResponse:
    concurrent_modification_exception: Optional[Any] = field(default=None)
    content_type: str = field(default=None)
    delete_application_input_processing_configuration_response: Optional[shared.DeleteApplicationInputProcessingConfigurationResponse] = field(default=None)
    invalid_argument_exception: Optional[Any] = field(default=None)
    invalid_request_exception: Optional[Any] = field(default=None)
    resource_in_use_exception: Optional[Any] = field(default=None)
    resource_not_found_exception: Optional[Any] = field(default=None)
    status_code: int = field(default=None)
    
